skyportal/handlers/api/shift.py

```diff
@@ -433,18 +431,17 @@ def patch(self, shift_id, user_id):
             # send a user notification to all members of the group associated to the shift
             # that the user needs to be replaced
             # recover all group users associated to the shift
-            shift = (
-                Shift.query_records_accessible_by(
+            try:
+                shift = Shift.get_if_accessible_by(
+                    [shift_id],
```

Comment:
you can drop the brackets and then you will get just the one shift, so you can also drop the `[0]` indexing below.